packages/ssx-sdk/src/ssx.ts

```diff
@@ -107,15 +113,93 @@ export class SSX {
   public storage: IStorage;
 
   constructor(private config: SSXConfig = SSX_DEFAULT_CONFIG) {
-    // TODO: initialize these based on the config
+    // TODO: pull out config validation into separate function
+    // TODO: pull out userAuthorization config
     this.userAuthorization = new UserAuthorization(config);
-    // get encryption config from config.modules.encryption
-    // determine which encryption module to use
-    // if encryption module is false, don't initialize encryption or dependent modules
-    this.encryption = new SignatureEncryption({}, this.userAuthorization);
-    this.dataVault = new BrowserDataVault({}, this.encryption);
-    this.credential = new Credential({}, this.dataVault);
-    this.storage = new BrowserStorage({});
+
+    // credential module disabled for now
+    // this.credential = new Credential({}, this.dataVault);
+
+    // initialize encryption module
+    // assume encryption module is **enabled** if config.encryption is not defined
+    const encryptionConfig =
+      config?.modules?.encryption === undefined
+        ? true
+        : config.modules.encryption;
+    if (encryptionConfig !== false) {
+      if (typeof encryptionConfig === 'object') {
+        // determine which encryption module to use
+        // initialize encryption with the provided config
+        if (encryptionConfig.module === 'SignatureEncryption') {
```

Comment:
The module field is prototyped with encryption as below
```ts
interface SSXEncryptionModuleConfig {
  module: 'SignatureEncryption' | 'LitEncryption';
}
```

There are specific types for each module that then has it's own specific config
```ts
interface SignatureEncryptionConfig extends SSXEncryptionModuleConfig {
  module: 'SignatureEncryption';
  /**
   * A message used to generate the detereministic signature for deriving the encryption key.
   * @default "Sign this message to generate an encryption key for {address}"
   */
  message?: () => string;
}
```

In this example, SignatureEncryption requires a configurable message, while LitEncryption (tbd) does not require this at all, but may instead require other config properties.

By specifying what module is used in this way, a developer can provide support for multiple, natively supported modules, while we provide a default.